src/webgpu/util/f32_interval.ts

```diff
@@ -169,15 +169,15 @@ export type F32Vector =
   | [F32Interval, F32Interval, F32Interval, F32Interval];
 
 /** Coerce F32Interval[] to F32Vector if possible */
-function isF32Vector(v: number[] | IntervalBounds[] | F32Interval[] | F32Vector): v is F32Vector {
+function isF32Vector(v: (number | IntervalBounds | F32Interval)[] | F32Vector): v is F32Vector {
   if (v[0] instanceof F32Interval) {
     return v.length === 2 || v.length === 3 || v.length === 4;
   }
   return false;
 }
 
 /** @returns an F32Vector representation of an array fo F32Intervals if possible */
-export function toF32Vector(v: number[] | IntervalBounds[] | F32Interval[] | F32Vector): F32Vector {
+export function toF32Vector(v: (number | IntervalBounds | F32Interval)[] | F32Vector): F32Vector {
```

Comment:
So you're now saying that `v` can be an array of any *mixed* element type of `number`, `IntervalBounds`, or `F32Interval`.

So, you could call `toF32Vector` with `[1, [2], [2,3], F32Interval.any()]`.

Please can we have some unit tests to check that this can correctly cope with the mix of element types?

Reply:
That is correct, a vector of things that is valid to toF32Interval should be convertible in toF32Vector. I have added tests for this.